website/content/en/preview/concepts/nodeclasses.md

```diff
@@ -244,6 +248,32 @@ kubelet:
   clusterDNS: ["10.0.1.100"]
 ```
 
+{{% alert title="Note" color="primary" %}}
+If you need to specify a field that isn't present in `spec.kubelet`, you can set it via custom [UserData]({{< ref "#specuserdata" >}}).
+For example, if you wanted to configure `maxPods` and `registryPullQPS` you would set the former through `spec.kubelet` and the latter through UserData.
+The following example achieves this with AL2023:
+
+```yaml
+apiVersion: karpenter.k8s.aws/v1
+kind: EC2NodeClass
+spec:
+  amiSelectorTerms:
+    - alias: al2023@latest
+  kubelet:
+    maxPods: 42
+  userData: |
+    apiVersion: node.eks.aws/v1alpha1
+    kind: NodeConfig
+    spec:
+      kubelet:
+        config:
+          # Configured through UserData since unavailable in `spec.kubelet`
+          registryPullQPS: 10
+```
+
+Note that when using the `Custom` AMIFamily you will need to specify fields **both** in `spec.kublet` and `spec.userData`.
```

Comment:
Does it matter the user is using Custom AMI Family?

Reply:
Yes, managed AMI families will have the fields automatically added by the generated UserData.